Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -1405,11 +1401,11 @@ TIFFImageIO::ReadGenericImage(void * _out, unsigned int width, unsigned int heig
 
     if (m_InternalImage->m_Orientation == ORIENTATION_TOPLEFT)
     {
-      image = out + (size_t)(row)*width * inc;
+      image = out + row * width * inc;
```

Comment:
VS2017 Code Analysis says:
> itktiffimageio.cxx(1404): warning C26451: Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow (io.2).

The problem introduced here is that `width * inc` is calculated in 32-bit, while on a 64-bit build, the multiplication was  previously calculated in 64-bit: `(size_t)(row)*width`.

A minimal fix could be to add a pair of parentheses:

    image = out + row * (width * inc);

`inc` is 64-bit size_t (on a 64-bit build), so then `(width * inc)` is calculated in 64-bit, and then `row * (width * inc)` is also calculated in 64-bit! What do you think? Anyway, I think this should be fixed one way or another before merging!

Reply:
> would it be worth mentioning in the commit message that your commit improves BitsPerSample support, as it now supports up to 63 bits per sample, instead of < 31 bits...?

I would say `uint64` type allows full (safe) support **palette** for 16-bit images (overflow if `int` is 16-bit only), and theoretical support for up to 64-bit palette (for what `m_TotalColors` is concerned), but for tiff above 16bit, an exception is raised elsewhere anyway. There may be other things to change/verify in the code so that it actually works, so I would not particularly insist on that...

> and then `row * (width * inc)`

or just move `inc` on the left !

Reply:
> overflow if int is 16-bit only

Does ITK support any platform that has 16-bit int? @dzenanz ?